test/Shared/UserManagerTestBase.cs

```diff
@@ -1165,19 +1174,19 @@ public Task<IdentityResult> ValidateAsync(UserManager<TUser> manager, TUser user
             var context = CreateTestContext();
             var manager = CreateManager(context);
             var roleManager = CreateRoleManager(context);
-            var role = CreateRole("addUserTest");
+            var roleName = "AddUserTest" + Guid.NewGuid().ToString();
```

Comment:
Hao was ok with having a flag in the method that treats supplied prefix as the actual value else we append the guid. The default is to append the guid. There a few tests that need to do it and this approach is simpler than having an `out` parameter. You can look at the api and let me know your thoughts